stdlib/prometheus/from.go

```diff
@@ -0,0 +1,587 @@
+			res = append(res, lm)
+			return res, nil
+		case ast.NotRegexpMatchOperator:
+			lm := &prompb.LabelMatcher{Type: prompb.LabelMatcher_NRE, Name: left, Value: right}
+			res = append(res, lm)
+			return res, nil
+		case ast.StartsWithOperator:
+			right = fmt.Sprintf("\"^%s.*\"", right)
+			lm := &prompb.LabelMatcher{Type: prompb.LabelMatcher_RE, Name: left, Value: right}
+			res = append(res, lm)
+			return res, nil
+		case ast.LessThanOperator:
+			return nil, errors.New("< not supported")
+		case ast.LessThanEqualOperator:
+			return nil, errors.New("<= not supported")
+		case ast.GreaterThanOperator:
+			return nil, errors.New("> not supported")
+		case ast.GreaterThanEqualOperator:
+			return nil, errors.New(">= not supported")
+		case ast.InOperator:
+			lm := &prompb.LabelMatcher{Type: prompb.LabelMatcher_RE, Name: left, Value: right}
+			res = append(res, lm)
+			return res, nil
+		default:
+			return nil, fmt.Errorf("unknown operator %v", n.Operator)
+		}
+	default:
+		return nil, fmt.Errorf("unsupported semantic expression type %T", n)
+	}
+}
+func (s *FromProcedureSpec) toLitteralMatcher(n semantic.Expression) (string, error) {
+	switch n := n.(type) {
+	case *semantic.StringLiteral:
+		return n.Value, nil
+	case *semantic.IntegerLiteral:
+		return fmt.Sprintf("%d", n.Value), nil
+	case *semantic.BooleanLiteral:
+		if n.Value {
+			return "true", nil
+		}
+		return "false", nil
+	case *semantic.FloatLiteral:
+		return fmt.Sprintf("%f", n.Value), nil
+	case *semantic.RegexpLiteral:
+		return n.Value.String(), nil
+	case *semantic.MemberExpression:
+		if n.Property == "_value" {
+			return "", errors.New("unable to push value filtering down to Prometheus")
+		}
+		return n.Property, nil
+	case *semantic.ArrayExpression:
+		vals := make([]string, 0, len(n.Elements))
+		for _, e := range n.Elements {
+			vals = append(vals, e.(*semantic.StringLiteral).Value)
+		}
+		return strings.Join(vals, "|"), nil
+	case *semantic.DurationLiteral:
+		return "", errors.New("duration literals not supported in storage predicates")
+	case *semantic.DateTimeLiteral:
+		return "", errors.New("time literals not supported in storage predicates")
+	default:
+		return "", fmt.Errorf("unsupported semantic expression type %T", n)
+	}
+}
+func (pi *PromIterator) Decode() (flux.Table, error) {
+	//maxLimit := int64(64)
+	if pi.spec.hasQuery {
+		if len(pi.result) > 0 {
+			return pi.ParseResult(pi.result[pi.index])
+		}
+		groupKey := execute.NewGroupKey(nil, nil)
+		builder := execute.NewColListTableBuilder(groupKey, &memory.Allocator{})
+		return builder.Table()
+	}
+	if len(pi.resultRemoteRead) > 0 {
+		return pi.ParseRemoteRead(pi.resultRemoteRead[pi.index])
+	}
+	groupKey := execute.NewGroupKey(nil, nil)
+	builder := execute.NewColListTableBuilder(groupKey, &memory.Allocator{})
+	return builder.Table()
+}
+
+// ParseResult convert Provider result to influx format
+func (pi PromIterator) ParseResult(series *QueryRangeResponseResult) (flux.Table, error) {
```

Comment:
The other methods on the PromIterator type are a pointer receivers, is there a reason this method needs to be a value receiver? If so can you comment as to why in the code? If not can you change it to be a pointer receiver otherwise calling the function makes a shallow copy of the PromIterator.

Reply:
There was no need, this PR is use now a Pointer